lib/matplotlib/axes/_axes.py

```diff
@@ -919,6 +921,63 @@ def axvline(self, x=0, ymin=0, ymax=1, **kwargs):
         self._request_autoscale_view(scalex=scalex, scaley=False)
         return l
 
+    @docstring.dedent_interpd
+    def axline(self, xy1, xy2, **kwargs):
+        """
+        Add an infinitely long straight line that passes through two points.
+
+        This draws a straight line "on the screen", regardless of the x and y
+        scales, and is thus also suitable for drawing exponential decays in
+        semilog plots, power laws in loglog plots, etc.
+
+        Parameters
+        ----------
+        xy1, xy2 : (float, float)
+            Points for the line to pass through.
+
+        Returns
+        -------
+        :class:`~matplotlib.lines.Line2D`
+
+        Other Parameters
+        ----------------
+        **kwargs
+            Valid kwargs are :class:`~matplotlib.lines.Line2D` properties,
+            with the exception of 'transform':
+
+            %(_Line2D_docstr)s
+
+        Examples
+        --------
+        Draw a thick red line passing through (0, 0) and (1, 1)::
+
+            >>> axline((0, 0), (1, 1), linewidth=4, color='r')
+
+        See Also
+        --------
+        axhline : for horizontal lines
+        axvline : for vertical lines
+        """
+
+        if "transform" in kwargs:
+            raise TypeError("'transform' is not allowed as a kwarg; "
+                            "axline generates its own transform")
+        x1, y1 = xy1
+        x2, y2 = xy2
+        line = mlines._AxLine([x1, x2], [y1, y2], **kwargs)
+        # Like add_line, but correctly handling data limits.
+        self._set_artist_props(line)
+        if line.get_clip_path() is None:
+            line.set_clip_path(self.patch)
+        if not line.get_label():
+            line.set_label(f"_line{len(self.lines)}")
+        self.lines.append(line)
+        line._remove_method = self.lines.remove
+        self.update_datalim([xy1, xy2])
```

Comment:
It might be nice to provide a way to exclude this

Reply:
To elaborate on this comment:

I've come across two use-cases where this scaling behavior is undesirable:
* I'm plotting a line with a single special point (eg, a ray, and I'm ok with the backwards portion of the line). If I plot `(origin, origin + direction)`, then my axis bounds scale to meet `origin + direction`, even though that point is arbitrary. I can work around that with a hack like `(origin, origin + eps*direction)`, but it forces me to trade off axis scaling against precision.

* I'm plotting a true infinite line, with no special points. Perhaps ideally I'd get axis scaling that shows the "nearest" part of the line, but I'd be ok with just passing `autoscale=False` to `axline` to have it skip this step.

Reply:
I agree that it should be possible to skip the autoscaling.  The two points being specified only need to be somewhere along the desired infinite line, and it's perfectly reasonable that they might not be within the domain one actually wants to view.  I'm actually wondering whether it would make more sense to leave out the `update_datalim` entirely instead of adding a kwarg; but maybe it's too late for that.  In any case, please open a new issue about this.

Reply:
Let's move this discussion to https://github.com/matplotlib/matplotlib/issues/16264 (which I replied to already).